File: ImageSmoother/ImageSmoother.py
class Solution:
    def imageSmoother(self, img: List[List[int]]) -> List[List[int]]:
        m = len(img)
        n = len(img[0])
        res = [[0] * n for i in range(m)]
        for i in range(m):
            for j in range(n):
                t = 0
                c = 0
                for dx in range(-1, 2):
                    for dy in range(-1, 2):
                        if 0 <= i + dx < m and 0 <= j + dy < n:
                            t += img[i + dx][j + dy]
                            c += 1
                res[i][j] = t // c
        return res

File: ImageSmoother/ImageSmoother.js
/**
 * @param {number[][]} img
 * @return {number[][]}
 */
var imageSmoother = function (img) {
  return img.map((arr, i) =>
    arr.map((_, j) => {
      var sum = 0;
      var count = 0;
      for (let l = i - 1; l <= i + 1; l++) {
        for (let m = j - 1; m <= j + 1; m++) {
          if (img[l] && img[l][m] > -1) {
            sum += img[l][m];
            count++;
          }
        }
      }
      return Math.floor(sum / count);
    })
  );
};
